Clarify the Identity Governance docs sidebar

ptgott opened the pull request:
Help readers form a mental model of the Identity Governance sidebar section and find relevant content. Make the following changes:

1. Move the Access Request Plugins section into the Access Requests section, ensuring that the top level of the Identity Governance sidebar section is a list of features. Shorten legacy file names beginning `ssh-approval`.

2. Move the use cases page first in the section, since it's a high-level conceptual guide.

3. Rename sidebar entries to remove redundant words and use the same grammatical form (e.g., using "-ing").

4. Add a subsection for IdP usage, allowing us to logically separate guides in the IdP section on setting up the IdP and guides about configuring the IdP to connect to external services.

ptgott committed:
Clarify the Identity Governance docs sidebar

Help readers form a mental model of the Identity Governance sidebar
section and find relevant content. Make the following changes:

1. Move the Access Request Plugins section into the Access Requests
   section, ensuring that the top level of the Identity Governance
   sidebar section is a list of features. Shorten legacy file names
   beginning `ssh-approval`.

2. Move the use cases page first in the section, since it's a high-level
   conceptual guide.

3. Rename sidebar entries to remove redundant words and use the same
   grammatical form (e.g., using "-ing").

4. Add a subsection for IdP usage, allowing us to logically separate
   guides in the IdP section on setting up the IdP and guides about
   configuring the IdP to connect to external services.

r0mant reviewed (commented):
I do like the overall structure and think it's cleaner, just a couple thoughts looking at the current organization.

- Can we better logically sort pages under Access Requests? It's weird that Automatic Reviews is first. Feels like role/resource requests should go first, then everything else.
- The "In Teleport Community Edition" reads weird as a title. Is it a typo?
- Under Plugins section, I would:
  - Move "How to build ..." to the bottom.
  - Move "Routing Access Request Notification" out of it and place it next to Automatic Reviews, or maybe even combine the two because they are basically describing how to use Access Monitoring Rules.
- For "Teleport as an IdP" - Azure and GCP guides are sort of buried under 3 levels of navigation and not easily discoverable, unless you know you need Teleport IdP for them. How can we make them more discoverable? Maybe cross-link to these guides from Enroll Resources / Applications / Securing Access to Cloud APIs as well?

<img width="284" height="719" alt="Image" src="https://github.com/user-attachments/assets/69b1bdfd-b8c1-471c-8270-f68e298a747c" />

ptgott committed:
Improve the Identity Governance docs organization

- In the Access Requests section, move Automatic Reviews below
  Role/Resource Requests. Also move Plugins before Configuration and
  Teleport Community Edition, since these guides are for a broader range
  of users.
- Fix the "Teleport Community Edition" label.
- Move "How to Build an Access Request Plugin" to the bottom of the
  subsection.
- Move Notification Routing Rules to the top level of Access Requests
- Mention the Teleport IdP, and link to the Usage subsection, in the
  introduction page for enrolling applications.

ptgott commented:
Thanks for taking a look!

> Can we better logically sort pages under Access Requests? It's weird that Automatic Reviews is first. Feels like role/resource requests should go first, then everything else.

I'll fix this.

> The "In Teleport Community Edition" reads weird as a title. Is it a typo?

This was supposed to read as "Access Requests In Teleport Community Edition", proceeding from the outer label to the inner one, but I can see how that doesn't really come through. I'll remove the "In".

> Under Plugins section, I would:
> Move "How to build ..." to the bottom.

I'll fix this as well.

> Move "Routing Access Request Notification" out of it and place it next to Automatic Reviews, or maybe even combine the two because they are basically describing how to use Access Monitoring Rules.

I'll move Routing Access Request Notification next to Automatic Reviews to limit the diff.

> For "Teleport as an IdP" - Azure and GCP guides are sort of buried under 3 levels of navigation and not easily discoverable, unless you know you need Teleport IdP for them. How can we make them more discoverable? Maybe cross-link to these guides from Enroll Resources / Applications / Securing Access to Cloud APIs as well?

Sounds good to me!
